internal/run/filters/filters.go

```diff
@@ -208,29 +233,48 @@ func byType(fs afero.Fs, vs []string, types []string) []string {
 	return vsf
 }
 
-func fillTypeMask(types []string) typeMask {
-	var mask typeMask
+func parseFileTypeFilter(types []string) fileTypeFilter {
+	var filter fileTypeFilter
 
 	for _, t := range types {
 		switch t {
 		case "executable":
-			mask |= typeExecutable
+			filter.simpleTypes |= typeExecutable
 		case "symlink":
-			mask |= typeSymlink
+			filter.simpleTypes |= typeSymlink
 		case "not executable":
-			mask |= typeNotExecutable
+			filter.simpleTypes |= typeNotExecutable
 		case "not symlink":
-			mask |= typeNotSymlink
+			filter.simpleTypes |= typeNotSymlink
 		case "binary":
-			mask |= typeBinary
+			filter.simpleTypes |= typeBinary
 		case "text":
-			mask |= typeText
+			filter.simpleTypes |= typeText
+		case "text/plain",
+			"text/html",
+			"text/xml",
+			"text/x-php",
+			"text/javascript",
+			"text/x-lua",
+			"text/x-perl",
+			"text/x-python",
+			"text/rtf",
+			"text/x-tcl",
+			"text/csv",
+			"text/tab-separated-values",
+			"text/vcard",
+			"text/calendar",
+			"text/vtt",
+			"text/x-shellscript",
+			"text/x-sh",
```

Comment:
If the intent is to support text/* types, maybe this could be refactored to do `strings.HasPrefix("text/")` so that when new such types are added to mimetype, they would not need to be separately added to lefthook?